Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/BayesQuasi.py

```diff
@@ -243,8 +242,15 @@ def PyExec(self):
         eProb = np.zeros(4 * nsam)
 
         group = ""
+        sample_workspace = AnalysisDataService.retrieve(self._samWS)
```

Comment:
Is there any way for this (or any of the ADS calls) to fail? Maybe loading in the data then clearing the ADS, then running?

Reply:
I think there is a possibility for it to fail if you load the workspace, click `Run`, and then while its running to quickly delete the workspace before it gets to this line in the algorithm, although its probably unlikely. 

It would probably be better to pass the workspace object in rather than the workspace name so we don't have to rely on the ADS. Do you think its worth making this change in this PR?

Reply:
I think it's very unlikely and you're maaking it no less safe then it was previously so lets just keep it like that.